libs/service-logging/src/service_logging/middleware.clj

```diff
@@ -0,0 +1,84 @@
+(ns service-logging.middleware
+  (:import [org.slf4j MDC])
+  (:use [slingshot.slingshot :only [try+ throw+]] )
+  (:require [clojure.string :as string]
+            [clojure.tools.logging :as log]
+            [clojure.data.json :as json]
+            [service-logging.thread-context :as tc]))
+
+(defn flatten-map
+  "Takes a nested map, and flattens it. The nested keys are combined with
+  the given separator, and the keys can optionally be pre-prended with a string"
+  [form separator pre]
+  (into {}
+        (mapcat (fn [[k v]]
+                  (let [prefix (if pre (str pre separator (name k)) (name k))]
+                    (if (map? v)
+                      (flatten-map v separator prefix)
+                      [[(keyword prefix) v]])))
+                form)))
+
+; TODO: Setup logger level conditional logging, which logs the body if the log
+;       level is at it's highest
+(defn- clean-request
+  [request]
+  (dissoc request
+          :body
+          :user-info
+          :compojure.api.middleware/options
+          :ring.swagger.middleware/data))
+
+(defn- clean-response
+  [response]
+  (dissoc response
+          :body))
+
+(defn log-request
+  [{:keys [request-method uri] :as request}]
+  (let [method (string/upper-case (name request-method))]
+    (tc/with-logging-context {:request (json/write-str (clean-request request))}
+                             (log/log 'AccessLogger :info nil (str method " " uri)))))
+
+(defn log-response
+  ([level throwable {:keys [request-method uri]} response]
+   (let [method (string/upper-case (name request-method))]
+     (tc/with-logging-context {:response (json/write-str (clean-response (assoc response
+                                                                           :uri uri
+                                                                           :request-method request-method)))}
+                              (log/log 'AccessLogger level throwable (str method " " uri)))))
+  ([request response]
+    (log-response :info nil request response))
+  ([level request response]
+   (log-response level nil request response)))
+
+(defn wrap-logging
+  "Logs incoming requests and their responses with the 'AccessLogger' logger.
+
+    If the response map contains a `throwable` key, the value will be given to
+    the logger as an exception/throwable. Also, if the `throwable` key is not nil,
+    it is assumed that there will also be an `exception` key in the response map.
+    The `throwable` and `exception` keys are not logged nor passed with the response."
+  [handler]
+  (fn [request]
+    (log-request request)
+    (let [response (handler request)
```

Comment:
I mentioned this in one of the commit diffs, but you can shorten this a little by altering the destructuring a little:

```clojure
(let [{:keys [throwable exception] :as response} (handler request)] ...)
```